File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/BrowseCouponsPositiveTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Optivem.Testing;

namespace SystemTests.Latest.AcceptanceTests;

public class BrowseCouponsPositiveTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task ShouldReturnPublishedCoupon(Channel channel)
    {
        await Scenario(channel)
            .Given().Coupon().WithCode("BROWSE10").WithDiscountRate(0.1m)
            .When().BrowseCoupons()
            .Then().ShouldSucceed()
            .And().Coupons().ContainsCouponWithCode("BROWSE10");
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/latest/acceptance/BrowseCouponsPositiveTest.java
package com.optivem.shop.systemtest.latest.acceptance;

import com.optivem.shop.systemtest.latest.acceptance.base.BaseAcceptanceTest;
import com.optivem.shop.dsl.channel.ChannelType;
import com.optivem.testing.Channel;

import org.junit.jupiter.api.TestTemplate;

class BrowseCouponsPositiveTest extends BaseAcceptanceTest {

    @TestTemplate
    @Channel(ChannelType.API)
    void publishedCouponShouldAppearInList() {
        scenario
                .given().coupon()
                    .withCode("BROWSE10")
                    .withDiscountRate(0.10)
                .when().browseCoupons()
                .then().shouldSucceed()
                    .coupons()
                        .containsCouponWithCode("BROWSE10");
    }
}
